pytorch_lightning/strategies/launchers/spawn.py

```diff
@@ -57,13 +57,10 @@ def launch(self, function: Callable, *args: Any, **kwargs: Any) -> Any:
         Arguments:
             function: The entry point for all spawned processes.
             *args: Optional positional arguments to be passed to the given function.
+            trainer: Optional reference to the :class:`~pytorch_lightning.trainer.trainer.Trainer` for which
+                a selected set of attributes get restored in the main process after processes join.
             **kwargs: Optional keyword arguments to be passed to the given function.
-                If a keyword argument named `trainer` is present and is an instance of
-                :class:`~pytorch_lightning.trainer.trainer.Trainer`, a selected set of attributes from the trainer get
-                restored in the main process after processes join. The `trainer` keyword argument will NOT be passed
-                into the function.
         """
-        trainer = kwargs.pop("trainer", None)
         os.environ["MASTER_PORT"] = str(self._strategy.cluster_environment.main_port)
```

Comment:
Also, if you don't mind:

```suggestion
        # fix the network port before spawning the processes. Otherwise, we would end up using a different port in
        # each spawned process because master port chooses a random free port and this would happen per process
        os.environ["MASTER_PORT"] = str(self._strategy.cluster_environment.main_port)
```